pkg/ipnet/zz_generated.deepcopy.go

```diff
@@ -0,0 +1,38 @@
+// +build !ignore_autogenerated
+
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by deepcopy-gen. DO NOT EDIT.
+
+package ipnet
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *IPNet) DeepCopyInto(out *IPNet) {
+	*out = *in
```

Comment:
This is a pretty incredible amount of complexity for this single line of code. Can we just use a simplified version of this implementation and drop all of the terrible k8s code generation?

Reply:
> Can we just use a simplified version of this implementation and drop all of the terrible k8s code generation?

I'm fine going that way to.  Auto-generation is at @dgoodwin's [gentle recommendation][1].  Auto-generation will help with scale as we grow the number of types which need these methods, but I'm fine with copy/paste manual generation even then ;).  @abhinavdahiya?

[1]: https://github.com/openshift/installer/issues/256#issuecomment-422492159

Reply:
> Can we just use a simplified version of this implementation...

In fact, it looks like `IPNet` will need a manual version of at least `DeepCopyInto` to [avoid][1]:

```
# github.com/openshift/installer/pkg/ipnet
pkg/ipnet/zz_generated.deepcopy.go:26:10: in.IPNet.DeepCopyInto undefined (type net.IPNet has no field or method DeepCopyInto)
```

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/276/pull-ci-openshift-installer-go-vet/294/build-log.txt

Reply:
can we do either completely manual or completely generated? Why is DeepCopyInto not generated automatically?

https://github.com/openshift/machine-config-operator/blob/master/Gopkg.toml#L70-L72
generates DeppCopyInto?? 
https://github.com/openshift/machine-config-operator/blob/master/pkg/apis/machineconfiguration.openshift.io/v1/zz_generated.deepcopy.go#L14-L20

Reply:
> can we do either completely manual or completely generated? Why is `DeepCopyInto` not generated automatically?

Completely manual is an option, although [there are trade-offs][1].  Completly automatic is not, see c3832d290b9dca073c373047dc217afc3925c15a.

[1]: https://github.com/openshift/installer/pull/276#discussion_r218605645